app/controllers/lti_nrps_controller.rb

```diff
@@ -84,11 +86,70 @@ def send_nrps_request
     lcd.last_synced = DateTime.current
     lcd.save
 
-    render json: members.as_json
+    # Update the roster with the retrieved set of members
+    @cuds = parse_members_data(lcd, members.as_json)
+    @sorted_cuds = @cuds.sort_by { |cud| cud[:color] || "z" }.reverse
+    @course = lcd.course
   end
 
 private
 
+  def parse_members_data(lcd, members_data)
+    cuds = CourseUserDatum.where(course_id: lcd.course_id, instructor: false,
+                                 course_assistant: false).to_set
+    email_to_cud = {}
+    cuds.each do |cud|
+      email_to_cud[cud.user.email] = cud
+    end
+
+    cud_view = []
+    members_data.each do |user_data|
+      next unless user_data["roles"].include? "http://purl.imsglobal.org/vocab/lis/v2/membership#Learner"
+
+      cud_data = {}
+      # Normalize email
+      user_data["email"] = user_data["email"].downcase
```

Comment:
consider `user_data["email"].downcase!` to do it in-place

Reply:
Will change

Reply:
![Screen Shot 2022-11-29 at 23 57 28](https://user-images.githubusercontent.com/25730111/204711120-8942efd9-d3c2-44e7-8290-c855c98e7f5a.png)
this change may have brought on an error

Reply:
nvm probably not this change specifically, will investigate

Reply:
<img width="926" alt="Screen Shot 2022-11-28 at 17 20 05" src="https://user-images.githubusercontent.com/25730111/204715364-0e450085-ed7d-4f6c-bfe3-8fca8f6a9cba.png">

So apparently a "Test Student" got added to the canvas course, and the test student doesn't have an email...

Reply:
We should add some code to ensure all fields we import into autolab exist

Reply:
Will add a check for this